### ISSUE TITLE
Control Plane does not asset `target` value when opening Contract Negotiation 

### ISSUE THREAD

**Issue by bscholtes1A** [eclipse-edc/Connector#4239]
# Bug Report

## Describe the Bug

When consumer initiates a contract negotiation, the control plane does not check if the content of the `target` field (which should contain the asset id) is equal to the asset id defined in the asset id portion of the policy `@id`.

In fact, the content of this `target` field is not even used, and 

### Expected Behavior

Control Plane should enforce that the content of the `target` field contains the same asset id as the one defined in the asset id portion of the policy `@id`, and return a validation error if it's not the case.

## Steps to Reproduce

Initiate a contract negotiation through the control plane management API, and use a random put a random string in the `target` field of the request.

**Comment by ndr-brt** [comment]
please note that the structure of the ContractOfferId (containing the assetId), is something EDC-related and used by the EDC provider to apply contract validations (see `ContractValidationServiceImpl`), but there could be a not-EDC-provider, in that case the ContractOfferId structure could be anything, so the consumer should not validate anything, is the provider that needs to do it (and it does already).

TL;DR: EDC provider in fact ignore the `target` attribute of the Offer because the `@id` already gives all the information needed
